pkg/chunkenc/memchunk.go

```diff
@@ -1733,21 +1742,44 @@ type sampleBufferedIterator struct {
 	extractor log.StreamSampleExtractor
 	stats     *stats.Context
 
-	cur        logproto.Sample
-	currLabels log.LabelsResult
+	cur        []logproto.Sample
+	currLabels []log.LabelsResult
 }
 
 func (e *sampleBufferedIterator) Next() bool {
+	if len(e.cur) > 1 {
+		e.cur = e.cur[1:]
+		e.currLabels = e.currLabels[1:]
+
+		return true
+	}
+
+	if len(e.cur) == 1 {
+		e.cur = e.cur[:0]
+		e.currLabels = e.currLabels[:0]
+	}
```

Comment:
This was very confusing for me. Could you add a comment what's happening here: element at index 0 is the current one and there are no more next elements, meaning that in order to load next batch, we need to clear the current slice `curr`